gulpfile.js

```diff
@@ -45,12 +45,18 @@ const parsePlatforms = () => {
         }
     }
 
-    // for osx and win, 32-bits works on 64, if needed
+    // for win, 32-bits works on 64, if needed
     if (availablePlatforms.indexOf('win64') === -1 && requestedPlatforms.indexOf('win64') !== -1) {
         validPlatforms.push('win32');
     }
-    if (availablePlatforms.indexOf('osx64') === -1 && requestedPlatforms.indexOf('osx64') !== -1) {
-        validPlatforms.push('osx32');
+    // for osx, only 64-bit nwjs exists. there is no 32-bit nwjs anymore,
+    // probably because Apple hasn't made 32-bit Macs for over a decade.
+    // related issue: https://github.com/nwjs/nw-builder/issues/363
+    // "The 32 bit version works but only has old legacy versions and it is
+    // not and probably won't be supported anymore." -- DblK @ nw-builder
+    // so trying to build as osx32 would fail since it has no new nwjs versions.
+    if (availablePlatforms.indexOf('osx32') === -1 && requestedPlatforms.indexOf('osx32') !== -1) {
+        validPlatforms.push('osx64');

```

Comment:
@xaiki This code means that if the user sends in --platforms=osx32 or tries to compile on 32-bit OSX (uhh does that even exist?), it will instead correct itself to building osx64.